web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -58,26 +58,21 @@ export class LayerDialogComponent implements OnDestroy {
   onSave() {
     let updatedLayers;
     if (this.layerId === ':new') {
-      updatedLayers = this.project.layers.set(
-        this.dataStoreService.generateId(),
-        {
-          id: this.dataStoreService.generateId(),
-        }
-      );
+      updatedLayers = {
+        [`layers.${this.dataStoreService.generateId()}`]: {},
```

Comment:
Nit: Could we make this more D.R.Y. by assigning id to a local var, e.g. 
`const layerId = this.layerId === ':new' ? this.dataStoreService.generateId() : this.layerId;`